src/state.rs

```diff
@@ -28,6 +28,19 @@ use crate::modules::ModuleConfig;
 use crate::modules::{compile_module, HostModules};
 use crate::{Schedule, VMError};
 
+#[derive(Debug, Clone)]
+pub struct Event {
+    pub origin: ContractId,
+    pub name: String,
+    pub data: Vec<u8>,
+}
+
+#[derive(Debug, Clone)]
+pub struct Receipt<R> {
+    pub ret: R,
```

Comment:
Instead of having a public `pub` field, implement `Deref<Target = R>` for Receipt.

Reply:
Makes a lot of sense. Done

Reply:
Still not done afaics

Reply:
No deref and ret is still public.